### PR TITLE
Add Landscape Mode for Tablet

### PR DESCRIPTION
**Pull request by noahmalmed**
### Changes:
- Update the app to be based completely on flex layout so the app can layout correctly even on page layout.
- Add an appState that updates the screen dimensions whenever there is a screen rotation.
- Fix an issue with the mardown library we use

# Review Checklist

- [x] Does it work in Android and iOS?
- [x] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [x] Are tests passing?

### CODE REVIEW

**Comment by wgranger** [comment]
Is there a way to trigger this in the simulator? I can't get the views to update to landscape when I rotate the display, which is how I'm assuming this should work.

File: src/constants/actions.js [hunk 1]
```diff
@@ -66,4 +66,6 @@ export const CLEAR_SHAPES = 'CLEAR_SHAPES'
 export const SAVE_EDITS = 'SAVE_EDITS'
 export const MUTATE_SHAPE = 'MUTATE_SHAPE'
 export const UNDO_EDIT = 'UNDO_EDIT'
-export const CLEAR_SHAPES_IN_PROGRESS = 'CLEAR_SHAPES_IN_PROGRESS'
```
Comment (wgranger) [inline src/constants/actions.js]: Is looks like `CLEAR_SHAPES_IN_PROGRESS` is still being used elsewhere in the code. Should it be removed there?

File: src/containers/zooniverseApp.js [hunk 2]
```diff
@@ -34,20 +36,30 @@ const mapDispatchToProps = (dispatch) => ({
   setNotificationPayload(value) {
     dispatch(setState('notificationPayload', value))
   },
-  setDimensions() {
-    dispatch(setDimensions())
-  },
```
Comment (wgranger) [inline src/containers/zooniverseApp.js]: `setDimensions` can be removed as a defined propType.

**Comment by noahmalmed** [comment]
> Is there a way to trigger this in the simulator? I can't get the views to update to landscape when I rotate the display, which is how I'm assuming this should work.

Because this PR makes changes to the native code, you may need to delete the app that's on your simulator and reinstall.

File: src/constants/actions.js (hunk 1 above)
Comment (noahmalmed) [reply src/constants/actions.js]: Oops, not sure why this got deleted!

**Comment by wgranger** [comment]
Yep. Deleting the app took care of the landscape switching. There's a slight bug with how the tutorial is rendered.

On viewing a project for the first time, the tutorial image and text is visible during each step of the tutorial. However, if you get into the project classifier and click on the "tutorial" tab, the image size grows and often pushes the text below view (you can scroll to view the text). Not sure why the image size would change from the initial render to the second time around.

**Comment by wgranger** [comment]
Nice. Takes care of the view on the tutorial. Merging.

**Pull request merged**